marabunta/model.py

```diff
@@ -281,27 +279,30 @@ def operation(self, exclude_addons=None):
 
 class Operation(object):
 
-    def __init__(self, command):
-        if isinstance(command, string_types):
-            command = self._shlex_split_unicode(command)
+    def __init__(self, command, shell=False):
+        """ Wrap a pexpect spawn command
+
+        :param command: the command to run as string
+        :param shell: boolean, when True, wraps the command in ``sh -c``
+                      so bash environment variables are interpolated
+        """
+        if not isinstance(command, string_types):
+            command = u' '.join(command)
         self.command = command
+        self.shell = shell
 
-    @staticmethod
-    def _shlex_split_unicode(command):
-        if sys.version_info < (3, 4):
-            return [l.decode('utf8') for l in shlex.split(
-                command.encode('utf-8'))]
+    def _spawn_command(self):
+        if self.shell:
+            return "sh -c '%s'" % (self.command,)
```

Comment:
What if `self.command` contains params with `'` ?

As for your example

`echo "backup command on ${DB_NAME}"`

What if I replace double quotes by single quotes

`echo 'backup command on ${DB_NAME}'`

Command will be incomplete.

```
└─> sh -c 'echo 'backup command on ${DB_NAME}''
backup
```

To escape char you will probably want to replace all `'` by `'"'"'` or `'\''` in self.command. (both use concatenation to close the quote, add an quote char and then reopen the quote) 

See https://stackoverflow.com/questions/1250079/how-to-escape-single-quotes-within-single-quoted-strings

This applies only to self.shell mode as there is no quote nesting.

Reply:
There is a better way :)

```python
>>> import pexpect
>>> import sys
>>> p = pexpect.spawn('sh', ['-c', 'echo "test test"'])
>>> p.logfile = sys.stdout
>>> p.expect(pexpect.EOF)
test test
0
>>> p.exitstatus
0
```

Reply:
Pushed as fixup commit